src/main/java/Controller/UserAPI.java

```diff
@@ -0,0 +1,120 @@
+import org.springframework.http.ResponseEntity;
+import javax.servlet.http.HttpServletRequest; 
+
+import java.util.List;
+import java.util.Map;
+
+@RestController
+public class UserAPI {
+
+    @Autowired 
+    private UserAccessor userAccessor;
+
+    @Autowired
+    private UniqueIDGenerator uniqueIDGenerator;
+
+    @Autowired
+    private MissingFields missingFields;
+
+    @Autowired
+    private ErrorResponseBody errorResponseBody;
+
+    @ExceptionHandler(UserAlreadyExistsException.class)
+    public ResponseEntity<Object> handleUserAlreadyExistsException(UserAlreadyExistsException e) {
+        return new ResponseEntity<Object>(errorResponseBody.getResponseBody(e), e.getHttpStatus());
+    }  
+     
+    @ExceptionHandler(UserRequiredFieldMissingException.class)
+    public ResponseEntity<Object> handleRequiredFieldMissingException(UserRequiredFieldMissingException e) {
+        return new ResponseEntity<Object>(errorResponseBody.getResponseBody(e), e.getHttpStatus());
+    }
+
+    @ExceptionHandler(InvalidLoginException.class)
+    public ResponseEntity<Object> handleInvalidLoginException(InvalidLoginException e) {
+        return new ResponseEntity<Object>(errorResponseBody.getResponseBody(e), e.getHttpStatus());
+    }
+
+    @ExceptionHandler(UserNotFoundException.class)
+    public ResponseEntity<Object> handleUserNotFoundException(UserNotFoundException e) {
+        return new ResponseEntity<Object>(errorResponseBody.getResponseBody(e), e.getHttpStatus());
+    }
+
+    @PostMapping(value = "/signup", consumes={"application/json"})
+    public Map<String, Object> signup(@RequestBody Map<String, Object> data, HttpServletRequest request) {
+
+        String path = request.getRequestURI();
+        String[] required = {"Username", "EmailID", "Password", "MobileNo"};
+        List<String> missing_fields = missingFields.getMissingFields(data, required);
+        if( missing_fields.size() > 0 ) {
+            throw new UserRequiredFieldMissingException(path, missing_fields);
+        }
+
+        /* necessary fields */
+        String username = data.get("Username").toString();
+        String emailID = data.get("EmailID").toString();
+        String password = data.get("Password").toString();
+        String mobileNo = data.get("MobileNo").toString();
+
+        /* optional field */
+        String base64Image = "";
+        if(data.get("Picture") != null){
+            base64Image = data.get("Picture").toString();
+        }
+
+        /* Checks if Username or email exists - throws exception if it does */
+        if (userAccessor.checkIfUserExists(username, emailID)) {
+            throw new UserAlreadyExistsException(path);
+        }
+        /* Generates unique userID */
+        long id = uniqueIDGenerator.generateUniqueID("User");
+        User newUser = new User(id, username, password, emailID, mobileNo, base64Image);
+        /* Inserts new entry into User table */
+        userAccessor.insert(newUser);
```

Comment:
generating unique id can be moved UserAccessor's insert method and that method can return this id.

Reply:
Done